mantidproject/mantid
# Patch IN16B diff detector reduction

**Description of work.**

Enables `IndirectILLReductionDIFF` to operate on runs even with alternating doppler profiles and acquisition modes.
Also does some code cleanup on the go.

**To test:**
Use data from cycle 203 experiment 7-03-189:
`313719:313775` 

Try different options: sum on vs off, transpose on vs off, and mask some components, like tube_8.
Also make sure that the individual runs are deleted, and only the final output is left (this wasn't the case before).



*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
